pex/common.py

```diff
@@ -464,10 +464,19 @@ def unlock():
             os.path.join(head, ".{}.atomic_directory.lck".format(tail or "here")),
             os.O_CREAT | os.O_WRONLY,
         )
-        # N.B.: Since lockf operates on an open file descriptor and these are guaranteed to be
-        # closed by the operating system when the owning process exits, this lock is immune to
-        # staleness.
-        fcntl.lockf(lock_fd, fcntl.LOCK_EX)  # A blocking write lock.
+        try:
+            # N.B.: Since lockf operates on an open file descriptor and these are guaranteed to be
+            # closed by the operating system when the owning process exits, this lock is immune to
+            # staleness.
+            fcntl.lockf(lock_fd, fcntl.LOCK_EX)  # A blocking write lock.
+        except OSError as e:
+            if e.errno == errno.EDEADLK:  # Resource deadlock avoided
```

Comment:
Do you understand the difference between this and `EDEADLOCK`? I don't.

Reply:
I found that on Ubuntu they are the same (both valued at 35). On MacOS only EDEADLK exists and the value is 11. So probably EDEADLK is more portable.

(I got here through the typeshed issue.)

Reply:
https://www.man7.org/linux/man-pages/man3/errno.3.html

>        EDEADLOCK
>              On most architectures, a synonym for EDEADLK.  On some
>              architectures (e.g., Linux MIPS, PowerPC, SPARC), it is a
>              separate error code "File locking deadlock error".